flake-module.nix

```diff
@@ -96,6 +96,25 @@ in
                 default = hp: { };
                 defaultText = ''Build tools useful for Haskell development are included by default.'';
               };
+              shell.extraHaskellLibraries = mkOption {
+                type = functionTo (types.listOf types.package);
+                description = ''
+                  A list of libraries that are included in the devShell, as if they were dependencies of one of the packages.
+
+                  This is useful for Haskell scripts that rely on the devShell to provide their libraries.
+                '';
+                default = p: [];
+                defaultText = lib.literalExpression "p: [ ]";
+                example = lib.literal
+              };
+              shell.buildInputs = mkOption {
+                type = types.listOf types.package;
+                description = ''
+                  A list of non-Haskell `buildInputs` that are included in the devShell.
+
+                  This is useful for dependencies on C libraries, such as `pkgs.zlib`.
```

Comment:
Is there an example Haskell project that does this? I assume that if a Haskell project depends on C library, we would have to add it in *two* places:

1. devShell's buildInputs
2. `addBuildDepends` on the local package itself

If so, wouldn't it be a nicer API to provide a single option that does both?

Emanote does this, but it is a binary dependency (rather than library dependency) for `stork`; (1) is [here](https://github.com/EmaApps/emanote/blob/cc65b533e780f902ca223268c93f5423c489182a/flake.nix#L37-L38) and (2) is [here](https://github.com/EmaApps/emanote/blob/cc65b533e780f902ca223268c93f5423c489182a/flake.nix#L50-L52).

Reply:
hercules-ci-agent does: https://github.com/hercules-ci/hercules-ci-agent/blob/master/nix/shellFor-cabal.nix

Progress converting it so far. No shell yet: https://github.com/hercules-ci/hercules-ci-agent/commit/e023859a16ac1727c55f06f067556fea6b57d5aa

> If so, wouldn't it be a nicer API to provide a single option that does both?

I agree, and this seems like something that shellFor should implement as well.